From 4a9866b17fb2a89f33751a57347bf705e4698462 Mon Sep 17 00:00:00 2001
Message-Id: <4a9866b17fb2a89f33751a57347bf705e4698462.1427265330.git.chang-joon.lee@intel.com>
In-Reply-To: <b4c66aa6a04d2be0fca9fa4448c7188a581a57a5.1427265330.git.chang-joon.lee@intel.com>
References: <b4c66aa6a04d2be0fca9fa4448c7188a581a57a5.1427265330.git.chang-joon.lee@intel.com>
From: Surbhi Kakarya <surbhi.kakarya@intel.com>
Date: Fri, 26 Dec 2014 12:32:42 +0530
Subject: [PATCH 12/22] FOR_UPSTREAM [VPG]: drm/i915: Fix for Hardware
 Rotation.

Current Implementation support HWRotation only on PIPE A.
This patch makes HW Rotation feature ready for each Pipe
i.e removing the pipe id check during Rotation

Issue: GMINL-4649
Change-Id: I8f302f8ef11540671bc64296a0a044bbb1a19bfd
Signed-off-by: Surbhi Kakarya <surbhi.kakarya@intel.com>
---
 drivers/gpu/drm/i915/intel_display.c |    2 +-
 drivers/gpu/drm/i915/intel_sprite.c  |    3 +--
 2 files changed, 2 insertions(+), 3 deletions(-)

diff --git a/drivers/gpu/drm/i915/intel_display.c b/drivers/gpu/drm/i915/intel_display.c
index 6a34a37..24009e0 100644
--- a/drivers/gpu/drm/i915/intel_display.c
+++ b/drivers/gpu/drm/i915/intel_display.c
@@ -2768,7 +2768,7 @@ static void i9xx_update_primary_plane(struct drm_crtc *crtc,
 		BUG();
 	}
 
-	if (intel_crtc->rotate180 && (pipe == 0))
+	if (intel_crtc->rotate180)
 		rotate = true;
 
 	if (INTEL_INFO(dev)->gen >= 4) {
diff --git a/drivers/gpu/drm/i915/intel_sprite.c b/drivers/gpu/drm/i915/intel_sprite.c
index aad82b37..edb61ef 100644
--- a/drivers/gpu/drm/i915/intel_sprite.c
+++ b/drivers/gpu/drm/i915/intel_sprite.c
@@ -525,8 +525,7 @@ vlv_update_plane(struct drm_plane *dplane, struct drm_crtc *crtc,
 				true, src_w != crtc_w || src_h != crtc_h);
 	}
 
-	if (intel_plane->rotate180 &&
-			(pipe == 0))
+	if (intel_plane->rotate180)
 		rotate = true;
 
 	/* Sizes are 0 based */
-- 
1.7.9.5

